build-support/bin/cherry_pick.sh

```diff
@@ -25,16 +25,30 @@ if [[ -z $PR_NUM ]]; then
   read -r PR_NUM
 fi
 
-MILESTONE=$2
-if [[ -z $MILESTONE ]]; then
-  MILESTONE=$(gh pr view "$PR_NUM" --json milestone --jq '.milestone.title')
-  if [[ -z $MILESTONE ]]; then
+TARGET_MILESTONE=$2
+if [[ -z $TARGET_MILESTONE ]]; then
+  TARGET_MILESTONE=$(gh pr view "$PR_NUM" --json milestone --jq '.milestone.title')
+  if [[ -z $TARGET_MILESTONE ]]; then
     echo "No milestone on PR. What's the milestone? (E.g. 2.10.x)"
-    read -r MILESTONE
+    read -r TARGET_MILESTONE
   fi
 fi
 
+# shellcheck disable=SC2016
+MILESTONES=$(gh api graphql -F owner=":owner" -F name=":repo" -f query='
+  query ListMilestones($name: String!, $owner: String!) {
+    repository(owner: $owner, name: $name) {
+      milestones(last: 10) {
+        nodes {title}
+      }
+    }
+  }' --jq .data.repository.milestones.nodes.[].title | awk "/$TARGET_MILESTONE/{p=1}p" -)
```

Comment:
I assume this is, in effect, the find "all milestones >= $TARGET_MILESTONE" logic? It's not obvious to me how this works, so a comment might be awesome!